docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -39,12 +39,34 @@ impl Storage {
 If you have defined a `Storage` struct following this naming scheme, then it will be made available to you through the reserved `storage` keyword within your contract functions.
 
 :::warning Using slot `0` is not supported!
-No storage values should be initialized at slot `0`. This is a known issue that will be fixed in the future.
+No storage values should be initialized at slot `0` - storage slots begin at `1`. This is a known issue that will be fixed in the future.
 :::
 
+## Storage Slots
+
+Each contract's state is represented with a storage slot key value store of type `Map<Field, Field>`. For now, storage slot positions for each variable must be explicitly assigned inside the `Storage impl`. Except for `Map` types, storage is in contiguous blocks, so you must calculate how many slots each public variable requires, and increment the next variable's slot by this amount.
```

Comment:
clarify this, mostly applies to public variables.  the private variable slot is in the bytecode, but doesnt show up at all in the public storage tree

Reply:
we dont need to calculate slots, its handled under the hood.  so everything just takes 1 slot

Reply:
some APIs take a storage slot as a parameter, see in `aztec-js` to help calculate the storage slot (like, if you want to add a note to a storage slot to a (private) map, a helper for doing the `hash(base_slot, key)` - helper to calculate this for you for a given contract address